src/GraphBLAS-sharp.Backend/Vector/SpMV.fs

```diff
@@ -0,0 +1,154 @@
+﻿namespace GraphBLAS.FSharp.Backend
+
+open Brahma.FSharp
+open GraphBLAS.FSharp.Backend
+open GraphBLAS.FSharp.Backend.ArraysExtensions
+open GraphBLAS.FSharp.Backend.Common
+open Microsoft.FSharp.Quotations
+
+module Vector =
+    let spMV
+        (clContext: ClContext)
+        (add: Expr<'c option -> 'c option -> 'c option>)
+        (mul: Expr<'a option -> 'b option -> 'c option>)
+        workGroupSize
+        =
+        //Until LocalMemSize added to ClDevice as member
+        let error = ref Unchecked.defaultof<ClErrorCode>
+
+        let localMemorySize =
+            Cl
+                .GetDeviceInfo(clContext.ClDevice.Device, OpenCL.Net.DeviceInfo.LocalMemSize, error)
+                .CastTo<int>()
+
+        let localArraySize1 = workGroupSize + 1
```

Comment:
Can it be reused in other algorithms to calculate the optimal size of the local array?